llvm/tools/sycl-post-link/SpecConstants.cpp

```diff
@@ -282,13 +476,22 @@ bool SpecConstantsPass::collectSpecConstantMetadata(
       Function *Callee = nullptr;
       if (!CI || CI->isIndirectCall() || !(Callee = CI->getCalledFunction()))
         continue;
-      std::pair<StringRef, unsigned> Res = getSpecConstMetadata(CI);
 
-      if (!Res.first.empty()) {
-        IDMap[Res.first] = Res.second;
-        Met = true;
+      if (Callee->getName().contains(SPIRV_GET_SPEC_CONST_COMPOSITE)) {
+        auto Res = getCompositeSpecConstMetadata(CI);
+        if (!Res.first.empty()) {
+          CompositeIDMap[Res.first] = Res.second;
+          Met = true;
+        }
+      } else if (Callee->getName().contains(SPIRV_GET_SPEC_CONST_VAL)) {
+        auto Res = getScalarSpecConstMetadata(CI);
+        if (!Res.first.empty()) {
+          ScalarIDMap[Res.first] = Res.second;
+          Met = true;
+        }
```

Comment:
These parts look like a copy-paste with small changes, although I didn't come up with a good suggestion to reduce repetitions.

Reply:
It seems to me that there is only ~3 similar lines here and this code snippet is only encountered once, so I don't think that this is a huge problem

Reply:
3/5 similar lines.